### ISSUE TITLE
o-forms: voice activation software does not work with implicit labels - ID: DAC_Dragon_Implicit_Labels_01

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#530]
# Issue
This is from page 86 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=inverse-form&system_code=origami-repo-data&brand=master>

Dragon voice activation software does not seem to recognise labels specified implicitly. This means that voice activation software users may be unable to reference components labelled in this way by name.

 
# Proposed Solution:
Add for=”required” to the label, this will allow Dragon users to reference the label by name.

```html
<label class=”o-forms-field o-forms-field—inverse” for=”required”> <span class=”o-forms-title”>
input</span>
input—invalid”> value=”” required=””> out this field</span>
<span class=”o-forms-title__main”>Required text </span>
<span class=”o-forms-input o-forms-input—text o-forms- <input id=”required” type=”text” name=”text” <span class=”o-forms-input__error”>Please fill
</span>
</label>
```

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=date&system_code=origami-repo-data&brand=master> and found on page 106 of the audit report

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=radio-box&system_code=origami-repo-data&brand=master> and found on page 112 of the audit report

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=radio-round&system_code=origami-repo-data&brand=master> and found on page 120 of the audit report

ID: DAC_Dragon_Implicit_Labels_04

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=toggle&system_code=origami-repo-data&brand=master> and found on page 120 of the audit report

ID: DAC_Dragon_Implicit_Labels_05

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=text-input&system_code=origami-repo-data&brand=master> and found on page 136 of the audit report

ID: DAC_Dragon_Implicit_Labels_06

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=select&system_code=origami-repo-data&brand=master> and found on page 148 of the audit report

ID: DAC_Dragon_Implicit_Labels_08

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=interactive&system_code=origami-repo-data&brand=master> and found on page 157 of the audit report

ID: DAC_Dragon_Implicit_Labels_09

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-table@9.0.2&demo=filter-input&system_code=origami-repo-data&brand=master> and found on page 195 of the audit report

ID: DAC_Dragon_Implicit_Labels_10

**Comment by notlee** [comment]
Cool, an easy if tedious fix
